lib/SPIRV/SPIRVLowerOCLBlocks.cpp

```diff
@@ -405,6 +405,28 @@ class SPIRVLowerOCLBlocks: public ModulePass {
       if (FirstArg->getName() != ".block_descriptor")
         continue;
 
+      bool isEnqueuedFunction = false;

```

Comment:
This function is too complicated. I'd like to separate this part as a new member function

  bool isEnqueuedFunction(Function &F);

and I think more strict checking should be done to identify if a function is enqueued function:
1. F is bitcasted to a pointer P
2. P is passed to a function call C of _Z21__spirv_EnqueueKernel

and add a comment about these two conditions that are used to decide if a function is enqueued function.

Thanks.